tests/chart_tests/test_vector.py

```diff
@@ -0,0 +1,294 @@
+
+
+@pytest.mark.parametrize(
+    "kube_version",
+    supported_k8s_versions,
+)
+class TestVector:
+    @staticmethod
+    def vector_common_tests(doc):
+        """Test common for vector daemonsets."""
+        assert doc["kind"] == "DaemonSet"
+        assert doc["metadata"]["name"] == "release-name-vector"
+
+    def test_vector_defaults_when_enabled(self, kube_version):
+        """Test that vector behaves as expected with default values when it is enabled."""
+        values = {"global": {"logging": {"collector": "vector"}}}
+        docs = render_chart(
+            kube_version=kube_version,
+            values=values,
+            show_only=all_templates,
+        )
+        assert len(docs) == 6
+        assert all(doc.get("apiVersion") for doc in docs)
+        assert all(doc.get("kind") for doc in docs)
+
+    @pytest.mark.skip("TODO: revisit this.")
+    def test_vector_daemonset_private_ca_certificates(self, kube_version):
+        """Test that helm renders a volume mount for private ca certificates for vector daemonset when private-ca-certificates are enabled."""
+        values = {
+            "global": {"logging": {"collector": "vector"}},
+            "privateCaCerts": ["private-root-ca"],
+        }
+        docs = render_chart(
+            kube_version=kube_version,
+            values=values,
+            show_only=["charts/vector/templates/vector-daemonset.yaml"],
+        )
+
+        get_containers_by_name(docs[0])
+        assert len(docs) == 1
+        pod_spec = docs[0]["spec"]["template"]["spec"]
+        assert len(pod_spec["volumes"]) == 2
+        assert len(pod_spec["containers"]) == 1
+
+        volume_mounts = [
+            {"name": "varlog", "mountPath": "/var/log/", "readOnly": True},
+            {"name": "config-volume-release-name-vector", "mountPath": "/etc/vector/config", "readOnly": True},
+        ]
+        assert pod_spec["containers"][0]["volumeMounts"] == volume_mounts
+
+    def test_vector_clusterrolebinding(self, kube_version):
+        """Test that helm renders a good ClusterRoleBinding template for vector when rbacEnabled=True."""
+        values = {"global": {"rbacEnabled": True, "logging": {"collector": "vector"}}}
+        docs = render_chart(
+            kube_version=kube_version,
+            values=values,
+            show_only=["charts/vector/templates/vector-clusterrolebinding.yaml"],
+        )
+
+        assert len(docs) == 1
+        doc = docs[0]
+        assert doc["kind"] == "ClusterRoleBinding"
+        assert doc["apiVersion"] == "rbac.authorization.k8s.io/v1"
+        assert doc["metadata"]["name"] == "release-name-vector"
+        assert len(doc["roleRef"]) > 0
+        assert len(doc["subjects"]) > 0
+
+        docs = render_chart(
+            kube_version=kube_version,
+            values={"global": {"rbacEnabled": False}},
+            show_only=["charts/vector/templates/vector-clusterrolebinding.yaml"],
+        )
+        assert len(docs) == 0
```

Comment:
This should be a different test. We can copy the above function definition and rename it to `_rbac_disabled`